File: Server/InvoiceHandling.Models/TaskDetail.cs
using System;

namespace InvoiceHandling.Models
{
	public class TaskDetail
	{
		public int Id { get; set; }
		public DateTime? DueDate { get; set; }
		public string PropertyHolderName { get; set; }
		public string PHAddress { get; set; }
		public string PHPhoneNumber { get; set; }
		public string AssignStatus { get; set; }
		public DateTime CreatedAt { get; set; }
		public int CallerSystemUserId { get; set; }
		public int JobCategoryId { get; set; }
		public decimal? MaximumBudget { get; set; }
		public int? ServiceProviderId { get; set; }
		public string Purpose { get; set; }
		public JobCategoryDetail JobCategoryDetail { get; set; }
		public ServiceProvider ServiceProvider { get; set; }
		public WorkOrderDetail WorkOrderDetail { get; set; }
	}
}

File: Client - CompayAdminPlatform/src/app/models/taskDetail.ts
import { CategoryDetail } from './categoryDetails';
import { WorkOrderDetail } from './workOrder';
import { ServiceProviderDetail } from './ServiceProviderDetail';

export interface TaskDetail {
  Id: number;
  DueDate: Date;
  PropertyHolderName: string;
  PHAddress: string;
  PHPhoneNumber: string;
  AssignStatus: string;
  CreatedAt: Date;
  Purpose: string;
  CallerSystemUserId: number;
  JobCategoryId: number;
  MaximumBudget: number;
  ServiceProvider: ServiceProviderDetail;
  WorkOrderDetail: WorkOrderDetail;
  JobCategoryDetail: CategoryDetail;
}
